tests/integrations/test_sonicmoe_fp4_cute_prep.py

```
@@ -0,0 +1,292 @@
+    torch.manual_seed(1)
+    x = torch.randn(4, 32)
+    packed, scale, pts = quantize_nvfp4_ref(x, per_tensor_scale=0.25)
+    assert float(pts) == 0.25
+    sim = dequantize_nvfp4_ref(packed, scale, pts)
+    bound = scale.float().repeat_interleave(SF_VEC_SIZE, dim=-1) * float(pts) + 1e-6
+    assert bool(((x - sim).abs() <= bound).all())
+
+
+def test_pack_scales_blocked_matches_naive_formula():
+    torch.manual_seed(2)
+    l, mn, sf_k = 2, 130, 6  # exercises both pad directions
```

Ruff (0.15.20)

[error] Ambiguous variable name: `l`

(E741)